README.md

```diff
@@ -3,21 +3,64 @@ README Languages:  [DE](README/README-DE.md) | [ES](README/README-ES.md) | [FR](
+The Real-time File Processing reference architecture is a general-purpose, event-driven, parallel data processing architecture that uses [AWS Lambda](https://aws.amazon.com/lambda). This architecture is ideal for workloads that need more than one data derivative of an object. 
+
+In this example application we deliver the notes from an interview in markdown format into S3 and we utilise CloudWatch Events to trigger multiple processing flows.
 
 ## Architectural Diagram
 
 ![Reference Architecture - Real-time File Processing](img/lambda-refarch-fileprocessing-simple.png)
 
+## Application Components
+
+### Event Trigger
+Unlike batch processing, in this architecture we process each individual file as it arrives. To achive this we utilise [CloudWatch Events](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/WhatIsCloudWatchEvents.html) and [CloudTrail](https://aws.amazon.com/cloudtrail/). We write a CloudWatch Events rule which checks for S3 PutObject API calls into our Source Bucket from CloudTrail. Everytime the PutObject API is called this creates a CloudTrail log which our rule translates into an event represented as a JSON object. In our rule we also define targets which our JSON event object is delivered to, which in this scenario is 4 seperate [SQS Queues](https://aws.amazon.com/sqs/) for 4 different worflows. Other target types include AWS Lambda Functions, Kinesis Data Streams, Simple Notification Service, Step Functions state machines, ECS tasks and more.
+
+### Conversion WorkFlow
+
+For this workflow the target of our JSON object decribing the S3 PutObject event is an SQS queue. Sending to SQS first rather than directly to Lambda allows for more control of Lambda invocations and better error handling.
+
+Lambda uses long polling to poll our queue and when messages are available it will read upto 5 batches and send them to our function. Lambda will automatically scale this process if more messages still remain on the queue. This allows us to simultaneously process upto 1000 batches.
+
+If our Lambda fails to process the messages we can configure SQS to send to a dead-letter queue for inspection and reprocessing.
+
+Once the function has the message this is parsed. The JSON event object contains information such as the S3 bucket and object key and object size.  
+
+Our business logic uses this information to retrieve the file from S3 and convert it to HTML finally writing out the new file to the HTML Bucket.
+
+
```

Comment:
Let's include a high-level representation of what the Lambda function is doing.  I like the explanation of why we're using the SQS queue but think we ought to trim down some of the other detail about how Lambda + SQS work.  Perhaps a link would suffice.

Reply:
I have added increased the lambda logic description and reduced the technical.